File: mycollab-web/src/main/java/com/esofthead/mycollab/module/user/accountsettings/view/parameters/RoleScreenData.java
package com.esofthead.mycollab.module.user.accountsettings.view.parameters;

import com.esofthead.mycollab.module.user.domain.Role;
import com.esofthead.mycollab.module.user.domain.criteria.RoleSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class RoleScreenData {
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<Role> {

		public Add(Role params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<Role> {

		public Edit(Role params) {
			super(params);
		}
	}

	public static class Search extends ScreenData<RoleSearchCriteria> {

		public Search(RoleSearchCriteria params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/user/accountsettings/view/parameters/RoleScreenData.scala
package com.esofthead.mycollab.module.user.accountsettings.view.parameters

import com.esofthead.mycollab.module.user.domain.Role
import com.esofthead.mycollab.module.user.domain.criteria.RoleSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.1.0
 */
object RoleScreenData {

  class Read(params:Integer) extends ScreenData[Integer](params) {}

  class Add(params:Role) extends ScreenData[Role](params) {}

  class Edit(params:Role) extends ScreenData[Role](params) {}

  class Search(params:RoleSearchCriteria) extends ScreenData[RoleSearchCriteria](params) {}
}
